### ISSUE TITLE
Intercepting device outputs - possible?

### ISSUE THREAD

**Issue by SuperV1234** [ValveSoftware/openvr#1394]
Let's say I want to create an overlay that allows me to adjust the angle of my controllers in games that have no native support for that. A possible idea would be intercepting the orientation returned by the devices themselves, tweaking it, and feeding it back to OpenVR so that games will get the new tweaked value instead of the raw orientation returned by the devices.

Is this possible with OpenVR? I've looked at the Wiki but couldn't find anything helpful.

**Comment by Raivr-dev** [comment]
Yes, I think this is possible as OpenVR-SpaceCalibrator does something similar (intercepting position and orientation, and changing that on the fly).
Source is here:
https://github.com/pushrax/OpenVR-SpaceCalibrator

**Comment by FreeTheSmoke** [comment]
Two very powerful options started from a user called _matzman666_ 

I suggest you check **OpenVR-InputEmulator** the fork by _procsynth_ is most up to date.
Description:
> An OpenVR driver that allows to create virtual controllers, emulate controller input, manipulate poses of existing controllers and remap buttons. A client-side library that communicates with the driver via shared-memory is also included.

Link https://github.com/procsynth/OpenVR-InputEmulator

Another very useful tool that I would recommend for anyone using VR is **OpenVR-Advanced-Settings**
Far too many awesome features to list
https://github.com/OpenVR-Advanced-Settings/OpenVR-AdvancedSettings#--space-offset-page
